File: java/javasearch/src/test/java/javasearch/SearchFileTest.java
package javasearch;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SearchFileTest {

    public SearchFileTest() {}

    @Test
    public final void test_searchfile_abs_path() {
        String path = "~/src/xsearch/java/javasearch/src/main/java/javasearch";
        String filename = "SearchFile.java";
        SearchFile searchFile = new SearchFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, searchFile.toString());
    }

    @Test
    public final void test_searchfile_rel_path1() {
        String path = ".";
        String filename = "SearchFile.java";
        SearchFile searchFile = new SearchFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, searchFile.toString());
    }

    @Test
    public final void test_searchfile_rel_path2() {
        String path = "..";
        String filename = "SearchFile.java";
        SearchFile searchFile = new SearchFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, searchFile.toString());
    }
}

File: kotlin/ktsearch/src/test/kotlin/ktsearch/SearchFileTest.kt
package ktsearch

import org.junit.Assert.assertEquals
import org.junit.Test
import java.io.File

class SearchFileTest {

    @Test
    fun test_searchfile_abs_path() {
        val path = "/Users/cary/src/xsearch/kotlin/ktsearch/src/main/kotlin/ktsearch/SearchFile.kt"
        val searchFile = SearchFile(File(path), FileType.CODE)
        assertEquals(path, searchFile.toString())
    }

    @Test
    fun test_searchfile_tilde_path() {
        val path = "~/src/xsearch/kotlin/ktsearch/src/main/kotlin/ktsearch/SearchFile.kt"
        val searchFile = SearchFile(File(path), FileType.CODE)
        assertEquals(path, searchFile.toString())
    }

    @Test
    fun test_searchfile_rel_path1() {
        val path = "./SearchFile.kt"
        val searchFile = SearchFile(File(path), FileType.CODE)
        assertEquals(path, searchFile.toString())
    }

    @Test
    fun test_searchfile_rel_path2() {
        val path = "../SearchFile.kt"
        val searchFile = SearchFile(File(path), FileType.CODE)
        assertEquals(path, searchFile.toString())
    }
}
